rust/ql/lib/codeql/rust/internal/PathResolution.qll

```diff
@@ -1501,8 +1521,17 @@ private ItemNode resolvePathCandQualified(
 ) {
   exists(string name, SuccessorKind kind |
     q = resolvePathCandQualifier(qualifier, path, name) and
-    result = getASuccessor(q, name, ns, kind) and
+    result = getASuccessor(q, name, ns, kind)
+  |
     kind.isExternalOrBoth()
+    or
+    // Non-public items are visible to paths in descendant modules of the declaring
+    // module; the declaration may happen via a `use` statement, where the item
+    // being used is _not_ itself in an ancestor module, and we currently don't track
+    // that information in `getASuccessor`. So, for simplicity, we allow for non-public
+    // items when the path and the item are in the same crate.
+    getCrate(path) = getCrate(result) and
+    not result instanceof TypeParam
```

Comment:
This line is just because the case we're handling here never overlaps with type parameters?

Reply:
It is to avoid the case that you originally fixed in https://github.com/github/codeql/pull/20096.